### PR TITLE
build: migrate to lerna

### PR DESCRIPTION
**Pull request by koonpeng**
## What's new

This completes the switch to lerna, the backend will continue to be npm@6 as it is still the default for node14, but the migration also fixes some compatibility issues with npm@7 so it maybe works now, but I have only tried it for `react-components`.

More importantly, this changes the workflow that we have currently, see the new readme for the new workflow, also take note of some caveats mentioned here https://github.com/open-rmf/rmf-web/tree/lerna#developing. 

## Self-checks

- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

**Review by matiasbavera — APPROVED** [review]
LGTM, did some testing and it's working correctly. Thanks for this. We may want to let the team know about these breaking changes.

File: packages/rmf-auth/package.json [hunk 1]
```diff
@@ -21,41 +22,49 @@
...
     "concurrently": "^5.3.0",
-    "@types/debug": "^4.1.5",
+    "eslint": "^7.26.0",
+    "eslint-plugin-react": "^7.21.4",
+    "eslint-plugin-react-hooks": "^4.1.2",
     "gh-pages": "^3.1.0",
-    "react-test-renderer": "^17.0.1",
-    "ts-node": "^9.0.0",
     "jest": "^26.5.0",
+    "react": "^16.13.1",
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/rmf-auth/package.json:54]: That is correct, one is a peer dependency which will not be automatically installed in npm < 7, the other is a dev dependency because we need to install it to run the tests.

File: pipenv-install/package.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "pipenv-install",
+  "version": "0.0.0",
+  "description": "dummy package",
+  "main": "index.js",
+  "scripts": {
+    "prepare": "node ../scripts/pipenv-install.js"
+  },
+  "author": "",
+  "license": "ISC"
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply pipenv-install/package.json:10]: I prefer to remove fields like `author` and `license` as this package will never be published.

File: packages/rmf-auth/.npmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+legacy-peer-dep=true
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/rmf-auth/.npmrc:1]: This is a left over when I was trying npm@7, mostly it was the deprecated packages like awesome-typescript-loader and istanbul-instrument-loader causing problems.
